src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -141,15 +141,15 @@ def _area_stats(self, payload):
                                              self.connected):
             return
         try:
-            device_stats = {k: v for k, v in self.device.stats.aggregated_stats.items()
-                            if v is not None}
-            market_stats = {k: v for k, v in self.market_area.stats.aggregated_stats.items()
-                            if v is not None}
+            device_stats = self.device.stats.aggregated_stats["bills"]
+            market_stats = self.market_area.stats.min_max_avg_rate_market(
```

Comment:
I am afraid there is a chance for concurrent access to the market stats, because this is executed in the Redis thread. Therefore if the stats are modified by the main thread while the Redis thread is reading them, an error will be raised. Despite that, let's not overengineer and wait for stress test to validate if I am right.

Reply:
Yes, I was thinking about deleting the _area_stats (and the corresponding list_(device)stats function on the d3a-api-client, because these numbers are pushed via the market_cycle event anyways. 
The min/max/avg prices can be acquired via the list_market_stats anyways. What do you think about this?

Reply:
@spyrostz

Reply:
@hannesdiedrich I agree, feel free to remove this. It makes sense IMO too, to read data directly from the market and not from the device, since we followed that approach.